posthog/hogql/database/database.py

```diff
@@ -413,8 +413,11 @@ def define_mappings(warehouse: dict[str, Table], get_table: Callable):
             events_join = next(
                 (
                     join
-                    for join in DataWarehouseJoin.objects.filter(team_id=team.pk).exclude(deleted=True)
-                    if join.joining_table_name == "events"
+                    for join in DataWarehouseJoin.objects.filter(
+                        team_id=team.pk,
+                        source_table_name=warehouse_modifier.table_name,
+                        joining_table_name="events",
+                    ).exclude(deleted=True)
```

Comment:
You can use `.first()` instead of using the loop and `next()` - it'll return `DataWarehouseJoin | None`